app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java

```diff
@@ -93,42 +91,20 @@ public void onClick(View v) {
         viewContacts.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                startActivity(new Intent(MainActivity.this, ContactListActivity.class));
+                Intent intent = new Intent(MainActivity.this, ContactListActivity.class);
+                intent.putExtra(ActivityConstantsUtils.SELECTED_PHONE_NUMBER, "");
+                startActivityForResult(intent, ActivityConstantsUtils.REQUEST_SELECTION_CONTACT);
             }
         });
 
     }
 
-    //---------------------------- STATIC METHODS TO SET selected contact FROM SMSContactRecyclerAdapter ----------
-
-    /**
-     * Return {@link SMSContact} selected contact whose address is shown in this view.
-     */
-    public static SMSContact getContactSelected(){
-        return contactSelected;
-    }
-
-    /**
-     * Set contact selected from contacts list.
-     *
-     * @param newContactSelected    {@link SMSContact} contact selected from contacts list.
-     */
-    public static void setContactSelected(SMSContact newContactSelected){
-        contactSelected = newContactSelected;
-    }
-
-    /**
-     * This method is invoked after an onBack of another activity.
-     * This is used to see if the phone number is been selected from contact list
-     * and the number is updated.
-     */
     @Override
-    protected void onResume() {
-        super.onResume();
-        if(contactSelected != null)
-            txtPhoneNumber.setText(contactSelected.getAddress());
-        else
-            txtPhoneNumber.setText("");
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if(resultCode == RESULT_OK && requestCode == ActivityConstantsUtils.REQUEST_SELECTION_CONTACT)
+            if(data.hasExtra(ActivityConstantsUtils.SELECTED_PHONE_NUMBER))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)